File: C++/001. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        if (n < 2) return {};
        unordered_map<int, int> hash;
        for (int i = 0; i < n; i++)
        {
            if (hash.count(target - nums[i]) > 0)
            {
                return {hash[target - nums[i]], i};
            }
            else
            {
                hash[nums[i]] = i;
            }
        }
        return {};
    }
};

File: python/001. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        visited = {}
        for i in range(len(nums)):
            if target-nums[i] in visited:
                res = [visited[target-nums[i]], i]
                return res
            else:
                visited[nums[i]] = i

